### PR TITLE
Windows: Add support for Python 3 build 

### LINKED ISSUE #16578: Windows: Package mantid with Python 3
Tasks:
- [x] Add Python 3 to 3rd party bundle
- [x] Rebuild Indirect fortran against Python 3
- [x] Build mantid against Python 3
- [x] Check user-bundle works

### PR DESCRIPTION
**Pull request by martyngigg**
**Description of work.**

This adds support for building against Python 3 on Windows. Python 3.8 has been added to the thirdparty bundle. Note that this does not switch the default build to Python 3. This will be done later once all other OSes fully support this.

⚠️ Please make sure mantidproject/thirdparty-msvc2015#50 is merged when this pull request is merged. ⚠️ 

**To test:**

Run cmake as usual but with the `WITH_PYTHON3` switch enabled.
Try out MantidPlot and Workbench and check that things *look* okay.

Fixes #16578. 

*This does not require release notes* because **it does not do anything visibly for users yet.**

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: qt/applications/workbench/workbench/app/mainwindow.py [hunk 1]
```diff
@@ -838,6 +846,7 @@ def main():
         # convert into absolute path
         options.script = os.path.abspath(os.path.expanduser(options.script))
         if not os.path.exists(options.script):
+            # TODO should be logged
```
Comment (DavidFair) [inline qt/applications/workbench/workbench/app/mainwindow.py]: This has been duplicated?

**Review by DavidFair — COMMENTED** [review]
One minor change with duplicating a todo

Otherwise PR works with a clean build on Windows, cleans up a lot and correctly starts

File: qt/applications/workbench/workbench/app/mainwindow.py (hunk 1 above)
Comment (martyngigg) [reply qt/applications/workbench/workbench/app/mainwindow.py]: Good catch. I must have slipped on the keyboard and pressed copy line!

**Comment by martyngigg** [comment]
@DavidFair I think I have fixed those warnings now.

**Comment by DavidFair** [comment]
The build hasn't finished yet - but we appear to still be getting warnings:
https://builds.mantidproject.org/job/pull_requests-windows/34081/consoleFull

**Comment by martyngigg** [comment]
It's never ending. I don't quite understand why all of these now show up as I haven't touched any of that stuff. I'll take a look.

**Comment by martyngigg** [comment]
@DavidFair I think I have it now :pray:

**Comment by DavidFair** [comment]
@martyngigg Alas, still no luck :cry:

**Comment by martyngigg** [comment]
I'm going to ask the builder to clean it so we can hopefully see all of the warnings in one shot.

**Review by DavidFair — APPROVED** [review]
Compiles without warnings (after a heroic effort by Martyn) and workbench and MantidPlot correctly start on Windows

Lets :shipit:

**Comment by martyngigg** [comment]
Thanks, David!

@mantidproject/gatekeepers Could we get this merged?

**Pull request merged**